src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -446,6 +451,35 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     return { hash };
   }
 
+  override async waitTransactionMined(hash: string): Promise<any> {
+    const web3Eth = this.getWeb3EthInstance();
+
+    return this.pollTransactionMined(hash, web3Eth);
```

Comment:
I don't fully understand why additional method was created. I don't see much benefit on waiting until previous `getTransactionReceipt` finishes fetching because each interval takes 6s and the fetch itself 100-200ms.

What do you think about this approach? (not tested, just rough idea)
```
return new Promise((resolve) => {
  let attemptNo = 0;
  const interval = setInterval(async () => {
    attemptNo++;

    <The logic with getTransactionReceipt>

    if (<Check transaction>) {
      clearInterval(interval);
      return resolve(transaction);
    } else if (attemptNo >= TRANSACTION_POLLING_MAX_ATTEMPTS) {
      clearInterval(interval);
      return resolve(null);
    }
  }, TRANSACTION_POLLING_INTERVAL);
});
```

Reply:
I like this approach, but I would still add some delay between calls so that we don't do 100 calls in case the polling takes 10s

Reply:
How 100 calls could happen? As far as I understand your code should try 10 times to ask the API if transaction is mined. The fetch action happens in 6 second delay. So all actions should take around 1 minute.